### ISSUE TITLE
Mypy allows calling `in` keyword on `Iterable`

### ISSUE THREAD

**Issue by tgavoille** [python/mypy#16532]
**Bug Report**

Mypy doesn't raise an error when checking if an element is in an `Iterable` while it does when directly using `__contains__`.

**To Reproduce**

```python

from typing import Any, Iterable

def iterable_contains(iterable: Iterable[Any], element: Any):
    if element in iterable:  # No error but I thingk it should raise
        print(element)
    if iterable.__contains__(element):  # Error, expected
        print(element)

```

**Expected Behavior**

Mypy should raise an error on the first if statement

**Actual Behavior**

But it doesn't

**Your Environment**

Found this with earlier python and mypy versions but example is reproducible with :

- Mypy version used: 1.7.0
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.11.0

**Comment by tgavoille** [comment]
Yes it seems so and actually [runtime implementation] https://docs.python.org/3/reference/datamodel.html#object.__contains__) states that `__contains__`  will try membership using `__iter__` then `__getitem__`. I'm sorry I didn't saw this before, probably would have been better as a discussion.

Could you point me to some documentation about how `in` might try other dunders than `__contains__` and why direct call to `__contains__` should fail if object implements `__iter__`

Thanks for the reply

**Comment by AlexWaygood** [comment]
Sure, it's documented here: https://docs.python.org/3/reference/expressions.html#membership-test-details

**Comment by tgavoille** [comment]
Thank you
